nilearn/glm/first_level/first_level.py

```diff
@@ -944,9 +952,12 @@ def first_level_from_bids(dataset_path, task_label, space_label=None,
                      img_specs[0])
 
     # Infer subjects in dataset
-    sub_folders = glob.glob(os.path.join(derivatives_path, 'sub-*/'))
-    sub_labels = [os.path.basename(s[:-1]).split('-')[1] for s in sub_folders]
-    sub_labels = sorted(list(set(sub_labels)))
+    if sub_labels:
```

Comment:
I'm a bit worried that there is no safety checks here. What if the user provides a list of subject labels that are not in the dataset? 
It woud be nice, for instance, to have a warning giving the list of labels which couldn't be found, and set `sub_labels` to the intersection between the existing and requested labels. WDYT?

Reply:
I think this has been addressed, right?